narwhal/types/src/primary.rs

```diff
@@ -315,6 +337,24 @@ impl From<HeaderDigest> for Digest<{ crypto::DIGEST_LENGTH }> {
     }
 }
 
+impl From<HeaderDigest> for Digest<{ crypto::INTENT_MESSAGE_LENGTH }> {
+    fn from(digest: HeaderDigest) -> Self {
+        let intent_message = IntentMessage::<HeaderDigest>::new(
+            Intent::default()
+                .with_scope(IntentScope::HeaderDigest)
+                .with_app_id(AppId::Narwhal),
+            digest,
+        );
+
+        Digest {
+            digest: bcs::to_bytes(&intent_message)
+                .expect("Serialization message should not fail")
+                .try_into()
+                .expect("INTENT_MESSAGE_LENGTH is correct"),
```

Comment:
Digest in fastcrypto is just a wrapper around &[u8], so im able to work around here to still use the same signature service interface (by append intent bytes and also initialize signature_service with length = digest_length + 3) . perhaps we want to rename it from Digest to something else.